lib/archethic/mining/distributed_workflow.ex

```diff
@@ -939,6 +978,86 @@ defmodule Archethic.Mining.DistributedWorkflow do
     })
   end
 
+  defp request_replication_validation(
+         context = %ValidationContext{transaction: tx, coordinator_node: coordinator_node},
+         node_public_key
+       ) do
+    storage_nodes = ValidationContext.get_chain_replication_nodes(context)
+
+    Logger.info(
+      "Send validated transaction to #{Enum.map_join(storage_nodes, ",", &Node.endpoint/1)} for replication's validation",
+      transaction_address: Base.encode16(tx.address),
+      transaction_type: tx.type
+    )
+
+    validated_tx = ValidationContext.get_validated_transaction(context)
+
+    message = %ValidateTransaction{
+      transaction: validated_tx
+    }
+
+    results =
+      Task.Supervisor.async_stream_nolink(
+        Archethic.TaskSupervisor,
+        storage_nodes,
+        fn node ->
+          {P2P.send_message(node, message), node}
+        end,
+        ordered: false,
+        on_timeout: :kill_task,
+        timeout: Message.get_timeout(message) + 2000
+      )
+      |> Stream.filter(&match?({:ok, _}, &1))
+      |> Stream.map(fn {:ok, {{:ok, res}, node}} -> {res, node} end)
+      |> Enum.to_list()
+
+    if Enum.all?(results, &match?({%Ok{}, _}, &1)) do
+      cross_validation_nodes = ValidationContext.get_confirmed_validation_nodes(context)
+
+      validation_nodes =
+        [coordinator_node | cross_validation_nodes]
+        |> P2P.distinct_nodes()
+        |> Enum.reject(&(&1.last_public_key == node_public_key))
+
+      Logger.info(
+        "Send replication validation message to validation nodes: #{Enum.map_join(validation_nodes, ",", &Node.endpoint/1)}",
+        transaction_address: Base.encode16(validated_tx.address),
+        transaction_type: validated_tx.type
+      )
+
+      new_context =
+        Enum.reduce(results, context, fn {_, %Node{first_public_key: first_public_key}}, acc ->
+          ValidationContext.add_replication_validation(acc, first_public_key)
+        end)
+
+      Task.Supervisor.async_stream(TaskSupervisor, results, fn {_,
+                                                                %Node{
+                                                                  first_public_key:
+                                                                    first_public_key
+                                                                }} ->
+        P2P.broadcast_message(validation_nodes, %NotifyReplicationValidation{
+          address: validated_tx.address,
+          node_public_key: first_public_key
+        })
+      end)
+      |> Stream.run()
+
+      new_context
+    else
```

Comment:
If there is a replication error returned by one of the storage node, the validation node enter in replication_error state. But the other validations nodes are waiting for this node to send the NotifyReplicationValidation message. So I think if there is a replication error, we should also notify other validation nodes that the sub shard is invalid and so other validation nodes can also enter in replication_error state. Actually they will just wait until the mining timeout of 1min is reached

Reply:
Yes, we can use the`ReplicationError` message, which is already handled by the process